File: 69-Sqrtx/69-Sqrtx.py
# Last updated: 07/08/2025, 19:32:36
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        k=0
        while k*k<x:
            k+=1
        if k*k==x: return k
        else : return k-1
        

File: 69-Sqrtx/69-Sqrtx.cpp
// Last updated: 07/08/2025, 19:32:29
class Solution {
public:
    int mySqrt(int x) {
        int st=0,end=x;
        while(st<=end){
            long mid = st+end >> 1;
            if (mid*mid==x) return mid;
            else if (mid*mid > x) end=mid-1;
            else st=mid+1;
        }
        return end;
    }
};